halo2_proofs/src/arithmetic.rs

```diff
@@ -189,79 +178,107 @@ fn serial_fft<G: Group>(a: &mut [G], omega: G::Scalar, log_n: u32) {
         r
     }
 
-    let n = a.len() as u32;
+    fn log2_floor() -> u32 {
```

Comment:
Was there a particular reason behind changing the original `log2_floor(num: usize)`?

Reply:
Yes.
`log2_floor` is not referred from other function so it should be local function as the same with `bitreverse`.
And the thread number can get where ever and it's not used in best_fft function so it should be local variable.
I think that makes more sense.
What do you think?

Reply:
Even though `log2_floor` is not used elsewhere, making it a standalone function would allow it to be tested separately, and IMHO is more understandable for review and maintenance. (In general, if you have a large complicated function like `*_fft` then moving the code of another function into it is going in the wrong direction.)

If `log2_floor` were kept as it is in this PR then it would need to be renamed, because it is no longer calculating the log2_floor of a general argument. But I prefer to revert this change.

Reply:
I would also prefer to move out `bitreverse` actually, but that should not be done in this PR.

Reply:
> If log2_floor were kept as it is in this PR then it would need to be renamed
making it a standalone function allows it to be unit-tested separately
I would also prefer to move out bitreverse actually, but that should not be done in this PR.

That makes sense.
I am going to revert.